apps/aecore/lib/aecore/channel/tx/channel_close_mutal_tx.ex

```diff
@@ -51,6 +50,19 @@ defmodule Aecore.Channel.Tx.ChannelCloseMutalTx do
   @spec get_chain_state_name :: atom()
   def get_chain_state_name, do: :channels
 
+  def chainstate_senders?(), do: true
```

Comment:
Could we deduce this from DataTx alone? For instance:
```
chainstate_senders?(data_tx), do: DataTx.senders(data_tx) == []
```

Reply:
Current way is safer. `chainstate_senders?` should depend on type. If due to some kind of bug no senders are included then SignedTx.validate can fail nicely.

I think your proposed solution although potentially easier to implement is too "hack-y". It's a little bit like "we encode the information that senders are from chainstate (and function to get them is expected in transaction module) by setting senders to `[]`". I think that's hacky.